cassandra/context.py

```diff
@@ -0,0 +1,57 @@
+# Copyright DataStax, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from cassandra.registry import MessageCodecRegistry
+from cassandra.protocol import ProtocolHandler
+
+__all__ = ['DriverContext']
+
+
+class SingletonProvider(object):
```

Comment:
It looks indeed like a LazyInitializer and it could be. However, I choosed to use a Provider for future needs. `Providers are strategies of accessing objects`.  We currently only have the SingletonProvider actually since we only need a single instance to be provided, but probably more provider types will be added when the DriverContext is more complete. e.g a FactoryProvider that always returns a new Connection instance based on the driver context and user configuration.

I guess this will be more clear in the future, so we can revisit this. Here is an interesting project related to the approach choosed for our pluggable driver:  http://python-dependency-injector.ets-labs.org. 

Let me know if you want we discuss more about this.